drivers/backup/utils/backup_utils.go

```diff
@@ -0,0 +1,275 @@
+	if err != nil {
+		debugMessage := fmt.Sprintf("config map: name [%s], namespace [%s]", configMapName, configMapNamespace)
+		return "", ProcessError(err, debugMessage)
+	}
+	kubeconfig, ok := cm.Data[kubeconfigKey]
+	if !ok {
+		err = fmt.Errorf("kubeconfig key [%s] not found in the config map [%s]", kubeconfigKey, configMapName)
+		return "", ProcessError(err)
+	}
+	err = os.WriteFile(filePath, []byte(kubeconfig), 0644)
+	if err != nil {
+		debugMessage := fmt.Sprintf("kubeconfig file: path [%s]; kubeconfig [%s]", filePath, kubeconfig)
+		return "", ProcessError(err, debugMessage)
+	}
+	return filePath, nil
+}
+
+// GetSourceClusterConfigPath returns the file path of the source cluster kubeconfig
+func GetSourceClusterConfigPath() (string, error) {
+	kubeconfigKeys := GetKubeconfigKeysFromEnv()
+	if len(kubeconfigKeys) < 2 {
+		err := fmt.Errorf("the environment variable KUBECONFIGS should have two kubeconfig keys: one for the source cluster and one for the destination cluster")
+		debugMessage := fmt.Sprintf("kubeconfig-keys: [%s]", kubeconfigKeys)
+		return "", ProcessError(err, debugMessage)
+	}
+	sourceClusterKubeconfigKey := kubeconfigKeys[0]
+	sourceClusterConfigPath, err := GetClusterConfigPath(sourceClusterKubeconfigKey, DefaultConfigMapName, DefaultConfigMapNamespace)
+	if err != nil {
+		debugMessage := fmt.Sprintf("source-cluster: kubeconfig-key [%s]", sourceClusterKubeconfigKey)
+		return "", ProcessError(err, debugMessage)
+	}
+	return sourceClusterConfigPath, nil
+}
+
+// GetDestinationClusterConfigPath returns the file path of the destination cluster kubeconfig
+func GetDestinationClusterConfigPath() (string, error) {
+	kubeconfigKeys := GetKubeconfigKeysFromEnv()
+	if len(kubeconfigKeys) < 2 {
+		err := fmt.Errorf("the environment variable KUBECONFIGS should have two kubeconfig keys: one for the source cluster and one for the destination cluster")
+		debugMessage := fmt.Sprintf("kubeconfig-keys: [%s]", kubeconfigKeys)
+		return "", ProcessError(err, debugMessage)
+	}
+	destinationClusterKubeconfigKey := kubeconfigKeys[1]
+	destinationClusterConfigPath, err := GetClusterConfigPath(destinationClusterKubeconfigKey, DefaultConfigMapName, DefaultConfigMapNamespace)
+	if err != nil {
+		debugMessage := fmt.Sprintf("destination-cluster: kubeconfig-key [%s]", destinationClusterKubeconfigKey)
+		return "", ProcessError(err, debugMessage)
+	}
+	return destinationClusterConfigPath, nil
+}
+
+// SwitchClusterContext switches the cluster context to the cluster specified by the clusterConfigPath
+//
+// SwitchClusterContext replicates the behaviour of the `tests.SetClusterContext` function in the common.go file of the tests package,
+// ensuring that errors encountered during the context switching process, including the case when retrieving the SSH node driver fails,
+// are appropriately processed using the ProcessError function and returned
+func SwitchClusterContext(clusterConfigPath string) error {
```

Comment:
why dont you fix SetClusterContext itself rather than duplicating the code

Reply:
To maintain consistency in error processing, I think it would be better for SwitchClusterContext to be included as part of the cluster controller package itself.